daiquiri/formatter.py

```diff
@@ -23,21 +25,138 @@
+    CHART_MOD = '38;5;%dm'  # 256-table foreground `38;5;00m`
+
+    LOG_LEVELS = (
+        logging.DEBUG,
+        logging.INFO,
+        logging.WARN,
+        logging.ERROR,
+        logging.CRITICAL,
+    )
+
+    def __init__(self, debug, info, warn, error, critical,
+                 base=False, name=None):
+        """Create a new color palette from console values.
+
+        Each argument (in `args`) corresponds to a logging level's color.
+
+        :param debug: color for debug level
+        :param info: color for info level
+        :param warn: color for warning level
+        :param error: color for error level
+        :param critical: color for critical level
+        :param base: (bool) use TTY defined colors instead of 256-color palette
+        """
+        self.name = name
+        self.base = base
+        self.colors = (debug, info, warn, error, critical)
+        self._level_colors = None
+
+    def __str__(self):
+        colors = (c[1] if not isinstance(c, int) else c for c in self.colors)
+        return '<Palette %s(%s)>' % (self.name if self.name else '',
+                                     ', '.join(map(str, colors)))
+
+    @property
+    def level_colors(self):
+        """Mapping of log levels to color prefixes.
+
+        :rtype: dict
+        """
+        # deferred processing, read only
+        if self._level_colors is None:
+            self._level_colors = self._process_colors()
+        return self._level_colors
+
+    def _process_colors(self):
+        """Process mapping of log levels.
+
+        :rtype: dict
+        """
+        ret = {}
+        # apply text formatting
+        for level, color in zip(self.LOG_LEVELS, self.colors):
+            if isinstance(color, tuple):
+                # text-decoration modifier
+                fmt, c_num = color
+            elif isinstance(color, int):
+                # no formatting
+                fmt, c_num = '0;%s', color
+            else:
+                raise ValueError(color)
+
+            if self.base:
+                c = self.BASE_MOD % c_num
+            else:
+                c = self.CHART_MOD % c_num
+            ret[level] = '%s%s' % (self.PREFIX, fmt % c)
+        return ret
+
+    @staticmethod
+    def bold(n):
+        """Bold color in Palette.
 
+        :param n: (int)
+        :rtype: tuple
+        """
+        return '1;%s', n
+
+    @staticmethod
+    def underline(n):
+        """Underline color in Palette.
+
+        :param n: (int)
+        :rtype: tuple
+        """
+        return '4;%s', n
+
+
+class SwatchMeta(type):
+    """Swatches type.
+
+    Used for attaching the palette name, defined in the Swatches
+    object, to the Palette instance.
+    """
+    def __new__(cls, name, bases, dct):
+        for k, v in dct.items():
+            if isinstance(v, Palette):
+                v.name = k
+        return super(SwatchMeta, cls).__new__(cls, name, bases, dct)
+
+
+class Swatches(six.with_metaclass(SwatchMeta, object)):
```

Comment:
I don't understand why `Swatch` as a name. Really.

I'd just go ahead with a module level definition

```python
PALETTE_DEFAULT = Palette(32, 36, Palette.bold(33), Palette.bold(31), Palette.bold(31), base=True)
# colors from the 256-color table
PALETTE_BLUE= Palette(39, 32, 25, 24, Palette.bold(23))
[...]
```
and that's good enough and simple.